src/main/java/org/occiware/tinom/extensions/ssh/SSHconnectionSensor.java

```diff
@@ -0,0 +1,153 @@
+	private static final String PRIVATE_KEY = "private.key";
+	private static final String USERNAME = "username";
+	private static final String PASSWORD = "password";
+	private static final String IPS_FILE = "ips.file";
+	private static final String SSH_PORT = "ssh.port";
+
+
+	/**
+	 * Validates a properties file content.
+	 * @param properties the properties
+	 * @throws Exception if an error occurred during the validation
+	 * */
+	public static void validate( Map<String,String> properties ) throws Exception {
+
+		Utils.checkKey(KNOWN_HOSTS, properties);
+		Utils.checkKey(PRIVATE_KEY, properties);
+		Utils.checkKey(USERNAME, properties);
+		Utils.checkKey(PASSWORD, properties);
+		Utils.checkKey(IPS_FILE, properties);
+		Utils.checkKey(SSH_PORT, properties);
+	}
+
+
+	/**
+	 * Connects to a remote host in ssh and executes a command.
+	 *
+	 * @param jsch used for ssh connection
+	 * @param host the name or the IP of the remote host
+	 * @param username a user to access to remote host
+	 * @param password the password associated to user
+	 * @param command a shell command to execute on remote host
+	 * @param port ssh port
+	 * @throws JSchException
+	 * */
+	public void connectAndExecuteCommand(JSch jsch, String host, String username, String password, String command, int port) {
+
+		// Session configuration
+		Session session = null;
+		Channel channel = null;
+		try {
+
+			session = jsch.getSession(username, host, port);
+			session.setPassword(password);
+			session.connect();
+
+			// Command execution
+			channel = session.openChannel("exec");
+			((ChannelExec)channel).setCommand(command);
+			channel.connect();
+
+		} catch (JSchException e) {
+
+			e.printStackTrace();
+		} finally {
+
+			session.disconnect();
+			channel.disconnect();
+		}
+	}
+
+
+	/**
+	 * Sends a script file to agents in ssh and executes it.
+	 * @param properties a properties file which indicates the agents IP location file
+	 * @param script a script to execute
+	 * @param appName an application name (useful when Tinom will integrate to Roboconf)
+	 * @throws Exception
+	 * */
+	public void sendAndExecuteScript(File properties, File script, String appName) throws Exception {
+
+		Properties prop = Utils.readPropertiesFile(properties);
+
+		// Validate properties file
+		Map<String,String> propMap = Utils.propertiesFileToMap(prop);
+		validate(propMap);
+
+		File agentsIP = new File((String) prop.get(IPS_FILE));
+		String knownHosts = (String) prop.get(KNOWN_HOSTS);
+		String privKey = (String) prop.get(PRIVATE_KEY);
+		String username = (String) prop.get(USERNAME);
+		String password = (String) prop.get(PASSWORD);
+		int port = Integer.parseInt( (String) prop.get(SSH_PORT) );
+
+		List<String> ips = Utils.readFileByLine(agentsIP);
+
+		// Prepare the command to execute
+		StringBuilder sb = new StringBuilder();
+		String scriptContent = Utils.readFileContent(script);
+
+		sb.append("bash <<ENDOFSCRIPT\n");
```

Comment:
Is bash really necessary?
How about executing the script on Windows as an example? Can't we directly pass the script's content to the SSH library?